gapis/shadertools/shadertools.go

```diff
@@ -208,15 +209,51 @@ func ExtractDebugSource(shader []uint32) (string, error) {
 
 	err := spvReflectErr(C.spvReflectCreateShaderModule(C.size_t(len(shader)*4), shaderPtr, &module))
 	if err != nil {
-		return "", err
+		return "", "", false, err
 	}
 	defer C.spvReflectDestroyShaderModule(&module)
 
+	sourceLanguage := C.GoString(C.spvReflectSourceLanguage(module.source_language))
+
 	if module.source_source == nil {
-		return "", errors.New("No Source File Detected")
+		var context C.spvc_context = nil
+		var ir C.spvc_parsed_ir = nil
+		var compiler C.spvc_compiler = nil
+		var options C.spvc_compiler_options = nil
+		var result *C.char = nil
+
+		C.spvc_context_create(&context)
+		C.spvc_context_parse_spirv(context, (*C.uint)(shaderPtr), (C.ulong)(len(shader)), &ir)
+		switch sourceLanguage {
+		case "GLSL":
+			if C.spvc_context_create_compiler(context, C.SPVC_BACKEND_GLSL, ir, C.SPVC_CAPTURE_MODE_TAKE_OWNERSHIP, &compiler) != C.SPVC_ERROR_INVALID_ARGUMENT {
```

Comment:
Here and below, also invert the condition and use an early exit to make it easier to read.